### PR TITLE
Enable Stigma functionality

### PR DESCRIPTION
**Pull request by gcwiak**

### CODE REVIEW

File: core/src/main/java/io/nixer/nixerplugin/core/stigma/evaluate/StigmaTokenFetchResult.java [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+ * Created on 2019-06-25.
+ *
+ * @author gcwiak
+ */
+public class StigmaTokenFetchResult {
+
+    @Nonnull
+    private final RawStigmaToken fetchedToken;
+
+    private final boolean originalTokenValid;
```
Comment (j-bron) [inline core/src/main/java/io/nixer/nixerplugin/core/stigma/evaluate/StigmaTokenFetchResult.java]: I am not sure we still need this information. Let's discuss offline.

File: core/src/main/java/io/nixer/nixerplugin/core/login/LoginActivityService.java [hunk 2]
```diff
@@ -5,7 +5,8 @@
 import io.nixer.nixerplugin.core.detection.rules.AnomalyRulesRunner;
 import org.springframework.util.Assert;
 
-public class LoginActivityService {
+// TODO find better name for this component
```
[the remark this answers is not in the document]
Comment (gcwiak) [reply core/src/main/java/io/nixer/nixerplugin/core/login/LoginActivityService.java]: I'm reluctant to use *DefaultXXX* naming convention as it does not bring any value.  This name is too generic and not helpful either. Also rules execution is to be gone from here in future changes so the only responsibility is going to be saving login attempts - I am thinking about renaming it to something like *LoginActivityStoringHandler*, but together with the future changes.

File: core/src/main/java/io/nixer/nixerplugin/core/stigma/evaluate/StigmaTokenService.java [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+            // TODO do we need this recording?
+            // covered by observe event or recording incoming_unreadable_token
+            // stigmaTokenStorage.recordLoginFailTokenInvalid(originalRawToken, stigmaTokenCheckResult.toString());
+
+            return new StigmaTokenFetchResult(newStigmaToken(), false);
+        }
+    }
+
+    @Nullable
+    private StigmaData tryObtainingStigmaData(@Nullable final RawStigmaToken originalToken) {
```
[the remark this answers is not in the document]
Comment (gcwiak) [reply core/src/main/java/io/nixer/nixerplugin/core/stigma/evaluate/StigmaTokenService.java]: I would stay with the current name as it does not contain implementation details.

File: core/src/main/java/io/nixer/nixerplugin/core/stigma/domain/StigmaStatus.java [hunk 4]
```diff
@@ -0,0 +1,14 @@
+package io.nixer.nixerplugin.core.stigma.domain;
+
+/**
+ * Represents possible statuses of stigma value.
+ */
+public enum StigmaStatus {
+    ISSUED,         // Stigma sent to proxy, but not yet used (not yet attached to HTTP traffic with set-cookie)
```
[the remark this answers is not in the document]
Comment (gcwiak) [reply core/src/main/java/io/nixer/nixerplugin/core/stigma/domain/StigmaStatus.java]: Ok, ISSUED status is now removed.

File: core/src/main/java/io/nixer/nixerplugin/core/stigma/evaluate/StigmaAction.java [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+        this.type = Preconditions.checkNotNull(type, "type");
+    }
+
+    @Nonnull
+    public RawStigmaToken getStigmaToken() {
+        return stigmaToken;
+    }
+
+    @Nonnull
+    StigmaActionType getType() {
```
[the remark this answers is not in the document]
Comment (gcwiak) [reply core/src/main/java/io/nixer/nixerplugin/core/stigma/evaluate/StigmaAction.java]: Agree, fixed.

File: core/src/main/java/io/nixer/nixerplugin/core/stigma/evaluate/StigmaTokenService.java [hunk 6]
```diff
@@ -0,0 +1,184 @@
...
+     * Consumes the currently used raw stigma token (might be null or empty) and returns a new token for further usage
+     * with information about validity of the original token.
+     */
+    @Nonnull
+    public StigmaTokenFetchResult fetchTokenOnLoginSuccess(@Nullable final RawStigmaToken originalToken) {
+
+        @Nullable final StigmaData stigmaData = tryObtainingStigmaData(originalToken);
+
+        if (isStigmaActive(stigmaData)) {
+            // TODO do we need this recording?
```
[the remark this answers is not in the document]
Comment (gcwiak) [reply core/src/main/java/io/nixer/nixerplugin/core/stigma/evaluate/StigmaTokenService.java]: Cleaned up, together with the other TODOs.

File: core/src/main/java/io/nixer/nixerplugin/core/stigma/storage/StigmaTokenStorage.java [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+import io.nixer.nixerplugin.core.stigma.domain.StigmaStatus;
+
+/**
+ * Created on 2019-06-12.
+ *
+ * @author gcwiak
+ */
+public interface StigmaTokenStorage {
+
+    void createStigma(Stigma stigma, final StigmaStatus status);
```
[the remark this answers is not in the document]
Comment (gcwiak) [reply core/src/main/java/io/nixer/nixerplugin/core/stigma/storage/StigmaTokenStorage.java]: Renamed

**Comment by gcwiak** [comment]
As discussed offline, the other comments are going to be addressed in a separate change set.

File: core/src/main/java/io/nixer/nixerplugin/core/login/LoginActivityService.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (j-bron) [reply core/src/main/java/io/nixer/nixerplugin/core/login/LoginActivityService.java]: Good point, the proposed name sounds great.

**Pull request merged**